setup.py

```diff
@@ -35,6 +38,43 @@ def run(self):
         os.system('rm -vrf ./build ./dist ./*.pyc ./*.tgz ./*.egg-info')
 
 
+def git_version(version):
+    try:
+        import git
+    except ImportError:
+        logger.warn('gitpython not found: Cannot compute the git version.')
+        return ''
+    try:
+        repo = git.Repo('.git')
+    except ImportError:
+        logger.warn('Git repo not found: Cannot compute the git version.')
+        return ''
+    sha = repo.head.commit.hexsha
+    if repo.is_dirty():
+        return '.dev0+{sha}.dirty'.format(sha=sha)
+    # commit is clean
+    # is it release of `version` ?
+    try:
+        tag = repo.git.describe(
+            match='[0-9]*', exact_match=True,
+            tags=True, dirty=True)
+        assert tag[1:] == version, (tag, version)
+        return ''
+    except git.GitCommandError:
+        return '.dev0+{sha}'.format(sha=sha)
```

Comment:
@aoen I lifted this logic from one of your pointers.  What the logic you would use for release vs release candidate?